wire-ios-sync-engine/Source/Calling/WireCallCenterV3.swift

```diff
@@ -478,23 +485,36 @@ extension WireCallCenterV3 {
 
         switch conversation.messageProtocol {
         case .mls:
-            return shouldEndCallForMLS(participants: participants)
+            return shouldEndCallForMLS(
+                previousParticipants: previousParticipants,
+                newParticipants: newParticipants)
         case .mixed, .proteus:
-            return shouldEndCallForProteus(participants: participants)
+            return shouldEndCallForProteus(
+                previousParticipants: previousParticipants,
+                newParticipants: newParticipants)
         }
     }
 
-    private func shouldEndCallForMLS(participants: [AVSCallMember]) -> Bool {
+    private func shouldEndCallForMLS(
+        previousParticipants: [AVSCallMember],
+        newParticipants: [AVSCallMember]
+    ) -> Bool {
         /// We assume that the 2nd participant is the other user, and if the other user's audio state is connecting, the call should end.
-        guard participants.count == 2,
-              participants[1].audioState == .connecting else {
+        guard
+            previousParticipants.count == 2,
+            newParticipants.count == 2,
+            newParticipants[1].audioState == .connecting
+        else {
```

Comment:
just wondering why the logic differs between MLS and Proteus (why don't we check that newPartipants.count == 1)

Reply:
@jullianm   I will try to answer without going to much into MLS details 😅
We are talking about 1:1 call. Here's an example:
`if Alice and Bob are calling in a 1:1 conversation over SFT, and Alice hangs up, Bob will receive callParticipantsChanged event from AVS.` (The more info about SFT you can find in the Confluence documentation.)
Now let's look at the differences between the protocols.
**Proteus:**
When Alice hangs up, SFT knows that the new amount of participants is 1 and sends this info to Bob.

**MLS:**
We use subgroups for calls. This means that Alice and Bob join a subgroup. When Alice hangs up, SFT still thinks the number of participants is 2 and sends this information to Bob. And Bob is responsible for leaving the subgroup and ending the call.

Reply:
got it, thanks! I will take a look at SFT doc to learn more about it 👍